integration-cli/docker_cli_wait_test.go

```diff
@@ -120,3 +121,43 @@ func TestWaitBlockedExitRandom(t *testing.T) {
 
 	logDone("wait - blocking wait with random exit code")
 }
+
+func TestWaitKilledContainerNoWait(t *testing.T) {
+	defer deleteAllContainers()
+
+	runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "/bin/cat")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		t.Fatal(out, err)
+	}
+	containerID := strings.TrimSpace(out)
+
+	killCmd := exec.Command(dockerBinary, "kill", containerID)
+	out, _, err = runCommandWithOutput(killCmd)
+	if err != nil {
+		t.Fatal(out, err)
+	}
+
+	done := make(chan struct{})
+	var wg sync.WaitGroup
```

Comment:
@pwaller the context was the same, just protect a goroutine (in the test) (I suppose)